firmware/robot2015/src-ctrl/modules/ControllerTaskThread.cpp

```diff
@@ -78,41 +79,34 @@ void Task_Controller(void const* args) {
             "MPU6050 not found!\t(response: 0x%02X)\r\n    Falling back to "
             "sensorless control loop.",
             testResp);
-
-        // Start a thread that can function without the IMU, terminate us if it
-        // ever returns
-        Task_Controller_Sensorless(mainID);
-
-        return;
     }
 
     // signal back to main and wait until we're signaled to continue
     osSignalSet(mainID, MAIN_TASK_CONTINUE);
     Thread::signal_wait(SUB_TASK_CONTINUE, osWaitForever);
 
-    const uint16_t ENC_TICKS_PER_TURN = 2048;
-
-    Pid motor2pid(0.0, 0.0, 0.0);
-
-    std::vector<int16_t> duty_cycles;
+    array<int16_t, 5> duty_cycles;
+    duty_cycles.fill(0);
 
-    const uint16_t kduty_cycle = 0;
-    duty_cycles.assign(5, kduty_cycle);
+    pidController.setPidValues(0.75, 0.2, 0);  // TODO: tune pid values
 
-    size_t ii = 0;
-    bool spin_rev = true;
-
-    int16_t duty_cycle_all = kduty_cycle;
+    // initialize timeout timer
+    commandTimeoutTimer = make_unique<RtosTimerHelper>(
+        [&]() { commandTimedOut = true; }, osTimerPeriodic);
 
     while (true) {
-        imu.getGyro(gyroVals);
-        imu.getAccelero(accelVals);
+        // imu.getGyro(gyroVals);
+        // imu.getAccelero(accelVals);
+
+        // note: the 4th value is not an encoder value.  See the large comment
+        // below for an explanation.
+        array<int16_t, 5> enc_deltas;
```

Comment:
Well, there is a small edge case where if duty_cycle errors for some reason and is too high, set_duty_get_enc would return early. Not very likely, but it feels in general just zero initializing things is safer as a default since it really doesn't change effeciency that much.